src/query/service/src/sql/planner/mod.rs

```diff
@@ -69,15 +70,24 @@ impl Planner {
         let mut tokenizer = Tokenizer::new(sql);
         if let Some(Ok(t)) = tokenizer.next() {
             if t.kind == TokenKind::INSERT {
-                if let Some(Ok(last_token)) = tokenizer.take(MAX_TOKEN_FOR_INSERT).last() {
-                    match self.plan_sql_inner(sql, Some(last_token.span.start)).await {
-                        Ok(v) => return Ok(v),
-                        Err(err) => {
-                            tracing::warn!(
-                                "Faster parser path for insert failed: {}, start to fallback",
-                                err
-                            );
+                let mut token_limit = MIN_TOKEN_FOR_INSERT;
+                while token_limit <= MAX_TOKEN_FOR_INSERT {
+                    let tokenizer = Tokenizer::new(sql);
```

Comment:
Do we need to create a new `Tokenizer` every time here?

Reply:
Yes, because it's consumed in the loop.